content/blog/2025-04-27-valkey-modules-rust-sdk-updates.md

```diff
@@ -0,0 +1,272 @@
++++
+title=  "Valkey Modules Rust SDK updates"
+date= 2025-04-27 01:01:01
+description= "Extending Valkey using Rust SDK."
+authors= ["dmitrypol"]
+categories= "modules"
++++
+
+## What are Valkey modules?  
+
+In an earlier [article](/blog/modules-101/) we discussed the process of building Valkey Modules.
+Modules allow adding extra features (such as new commands and data types) to Valkey without making changes to the core code.
+We briefly touched on how to use [Valkey Modules Rust SDK](https://github.com/valkey-io/valkeymodule-rs) to build a simple module.
+In this article we will expand on this topic and discuss some of the new features added to the SDK over the last year.
+SDK itself also provides examples on how to use these new features but this article summarizes them below.
+
+## What is Valkey Modules Rust SDK?  
+
+The SDK is based on [Redis Modules Rust SDK](https://github.com/RedisLabsModules/redismodule-rs) and provides abstraction APIs on top of Valkey Modules own API.
+For those familiar with Rust development the SDK is a regular Rust crate that can be added to Cargo.toml file like any other Rust dependency.
+It requires functionality to be present in the underlying Valkey version but allows writing Valkey modules in Rust, without needing to use raw pointers or unsafe code.
+Recently released [Bloom Filers module](/blog/introducing-bloom-filters/) is built using this crate and several of the developers who worked on the module contributed to the SDK.
+Let's deep dive into to new features.
+
+## Client
+
+`Context` struct now has several new functions to get information about the client.
```

Comment:
'client' tends to be a overloaded term. I think this is about the client connected to Valkey, right? Maybe just be a little more percise.

Reply:
updated